tests/test_rocm_sanity.py

```diff
@@ -51,21 +58,36 @@ def test_rocm_output(self, rocm_info_output, to_search):
 
     def test_hip_printf(self):
         # Compiling .cpp file using hipcc
+        hipcc_executable = "./hipcc" if PLATFORM == "linux" else "hipcc.exe"
```

Comment:
FYI Windows will run `program_foo.exe` if you just say `program_foo`. What I think doesn't work here is if you set `-o` to a file name without the `.exe` extension, because Windows doesn't have the same "every file is executable" concept that Linux does. So since `hipcc.exe` already exists, you should be able to use `hipcc` throughout, but you still need the suffix for `hipcc_check`.

Another pattern you could use is
```python
platform_executable_suffix = ".exe" if platform.system() == "Windows" else ""
hipcc_check_executable = f"hipcc_check{platform_executable_suffix}"
```

As needed, you could do the same thing with the `./` prefix. That way, there would be only one spot near the start of the test that handles platform differences.